version: '3.8'

services:
  bot:
    build:
      context: .
    volumes:
      - .:/bot
    command: python bot/bot.py
    environment:
      RABBITMQ_HOST: rabbitmq
      DATABASE_URL: postgres://$POSTGRES_USER:$POSTGRES_PASSWORD@database:5432/$POSTGRES_DB


  worker:
    build:
      context: .
      dockerfile: workers/Dockerfile.worker
    depends_on:
      - rabbitmq
      - database
    environment:
      RABBITMQ_HOST: rabbitmq
      DATABASE_URL: postgres://$POSTGRES_USER:$POSTGRES_PASSWORD@database:5432/$POSTGRES_DB


  rabbitmq:
    image: rabbitmq:3.9-management
    ports:
      - "15672:15672"
      - "5672:5672"
    volumes:
      - rabbitmq_volume:/var/lib/rabbitmq
    restart: on-failure

  database:
    image: postgres:13.3
    volumes:
      - postgres_volume:/var/lib/postgresql/data
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    restart: always

